docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+);
+----
++
+. Add a new metrics endpoint.
++
+[source,java]
+----
+final Micrometer micrometer = new Micrometer();
+app.get("/metrics", ctx -> {
+  ctx.status(404);
+  if (ctx.basicAuthCredentialsExist()) {
+    final BasicAuthCredentials credentials = ctx.basicAuthCredentials();
+    if ("metrics".equals(credentials.getUsername()) && "secret".equals(credentials.getPassword())) {
+      ctx.status(200).result(micrometer.scrape());
+    }
+  }
+});
+----
++
+The `MicroMeter` class here is a self-written class that sets up a
+couple of metrics monitor, and also creates the registry
+for Prometheus, which in turn provides the text-based Prometheus output.
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.jvm.JvmCompilationMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmHeapPressureMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.logging.Log4j2Metrics;
+import io.micrometer.core.instrument.binder.system.FileDescriptorMetrics;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import io.micrometer.core.instrument.binder.system.UptimeMetrics;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+
+public class Micrometer {
+
+    final PrometheusMeterRegistry registry = new PrometheusMeterRegistry(new PrometheusConfig() {
+        @Override
+        public String get(String key) {
+            return null;
+        }
+
+        @Override
+        public String prefix() {
+            return "javalin";
+        }
+    });
+
+    public Micrometer() {
+        Metrics.addRegistry(registry);
+        new JvmGcMetrics().bindTo(Metrics.globalRegistry);
+        new JvmHeapPressureMetrics().bindTo(Metrics.globalRegistry);
+        new JvmThreadMetrics().bindTo(Metrics.globalRegistry);
+        new JvmCompilationMetrics().bindTo(Metrics.globalRegistry);
+        new JvmMemoryMetrics().bindTo(Metrics.globalRegistry);
+        new Log4j2Metrics().bindTo(Metrics.globalRegistry);
+        new UptimeMetrics().bindTo(Metrics.globalRegistry);
+        new FileDescriptorMetrics().bindTo(Metrics.globalRegistry);
+        new ProcessorMetrics().bindTo(Metrics.globalRegistry);
+    }
+
+    public String scrape() {
+        return registry.scrape();
+    }
+}
+----
++
+. Restart your app and poll the metrics endpoint.
++
+[source,bash]
+----
+curl localhost:7000/metrics -u metrics:secret
+----
++
+This returns a line based response with one metric per line. This is the
+standard Prometheus format.
+
+[discrete]
+==== Install and configure {metricbeat}
+
+. Time to install and configure {metricbeat}.
+For details, see <<install-metricbeat,Install {metricbeat}>>.
```

Comment:
I think we should provide the instructions inline (as I've suggested for Filebeat) so users don't have to change contexts.